lib/VMwareWebService/broker_timeout.rb

```diff
@@ -0,0 +1,64 @@
+module Timeout
```

Comment:
Ok, comments aside, can we :+1: merging this for now @roliveri @matthewd @tenderlove?

I think we all agree that the current behavior of ruby 2.0's VM is causing us to need to guard the "critical section" of Timeout.rb.  We need to continue to investigate this on ruby 2.1.6 (since it's 99% or more compatible with our app) and try to fix the VM issue or Timeout.rb upstream depending on where it makes sense to fix it.  I can easily setup some test appliances to recreate this on ruby 2.1.6 without @roliveri's rewrite of Timeout.rb.

My concern is we don't know what we don't know so we can't really know if Timeout.rb or the ruby's Thread kill/raise is the problem.  Let's decide on that when we have more information.

thoughts?